### PR TITLE
Add Nim client code generator

### PR DESCRIPTION
**Pull request by hokamoto**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

I add the client code generator for [Nim Programming Language](https://nim-lang.org/).

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/NimClientCodegen.java [hunk 1]
```diff
@@ -0,0 +1,309 @@
+package org.openapitools.codegen.languages;
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/NimClientCodegen.java]: I'll add the copyright header in another PR.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/NimClientCodegen.java [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+  public CodegenType getTag() {
+    return CodegenType.CLIENT;
+  }
+
+  public String getName() {
+    return "nim";
+  }
+
+  public String getHelp() {
+    return "Generates a nim client.";
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/NimClientCodegen.java]: I'll add the word "beta" as follows:

`Generates a nim client (beta).`

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/NimClientCodegen.java [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+    name = name.replace(" ", "_");
+    name = StringUtils.camelize(name, false);
+
+    if (name.matches("\\d.*")) { // starts with number
+      return "`" + name + "`";
+    } else {
+      return name;
+    }
+  }
+}
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/NimClientCodegen.java]: We're using 4 space indention so I'll reformat the code in a separate PR.

**Comment by wing328** [comment]
> @wing328 can we merge this, as it is a new generator, or does anyone else have to review it?

I left a couple of comments. Will work on those in a separate PR.

Let's launch this and see if nim developers have feedback on this new generator.

**Comment by wing328** [comment]
@hokamoto thanks for the new generator. may I know if you've a Twitter account? 

We'll tag you in the tweet to promote the new generator.

**Comment by hokamoto** [comment]
@wing328 Hi, my Twitter account is @tox2ro (https://twitter.com/tox2ro)

**Comment by wing328** [comment]
Thanks. Here is the Tweet: 

https://twitter.com/oas_generator/status/1172492657999794176

**Comment by wing328** [comment]
@hokamoto thanks for the PR, which has been included in the v4.1.3 release: https://twitter.com/oas_generator/status/1180123829626003456

**Pull request merged**